sinatra-contrib/lib/sinatra/required_params.rb

```diff
@@ -60,7 +62,7 @@ def _required_params(p, *keys)
         elsif key.is_a?(Array)
           _required_params(p, *key)
         else
-          halt 400 unless p && p.respond_to?(:has_key?) && p.has_key?(key.to_s)
+          halt 400 unless p.respond_to?(:has_key?) && p&.key?(key.to_s)
```

Comment:
Here `has_key?` was replaced by `key?` however we still ask if `p` responds to `has_key?`

```suggestion
          halt 400 unless p.respond_to?(:key?) && p&.key?(key.to_s)
```

Reply:
Good suggestion 👍